ocs_ci/ocs/ripsaw.py

```diff
@@ -0,0 +1,125 @@
+
+
+class RipSaw(object):
+    """
+      Workload operation using RipSaw
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function
+
+        Args:
+            kwargs (dict):
+                Following kwargs are valid
+                repo: Ripsaw repo to used - a github link
+                branch: branch to use from the repo
+                namespace: namespace for the operator
+
+        Example Usage:
+            r1 = RipSaw()
+            r1.apply_crd(crd='ripsaw_v1alpha1_ripsaw_crd.yaml')
+            # use oc apply to apply custom modified bench
+            my_custom_bench = my_custom_bench.yaml
+            run_cmd('oc apply -f my_custom_bench')
+        """
+        self.args = kwargs
+        self.repo = self.args.get('repo', 'https://github.com/cloud-bulldozer/ripsaw')
+        self.branch = self.args.get('branch', 'master')
+        self.namespace = self.args.get('namespace', 'ripsaw')
+        self.pgsql_is_setup = False
+        self.ocp = OCP()
+        self.ns_obj = OCP(kind='namespace')
+        self._create_namespace()
+        self._clone_ripsaw()
+
+    def _create_namespace(self):
+        """
+        create namespace for RipSaw
+        """
+        self.ocp.new_project(self.namespace)
+
+    def _clone_ripsaw(self):
+        """
+        clone the ripaw repo
+        """
+        self.dir = tempfile.mkdtemp(prefix='ripsaw_')
+        try:
+            log.info(f'cloning ripsaw in {self.dir}')
+            git_clone_cmd = f'git clone -b {self.branch} {self.repo} '
+            run(
+                git_clone_cmd,
+                shell=True,
+                cwd=self.dir,
+                check=True
+            )
+            self.crd = 'resources/crd/'
+            self.operator = 'resources/operator.yaml'
+        except (CommandFailed, CalledProcessError)as cf:
+            log.error('Error during cloning of ripsaw repository')
+            raise cf
+
+    def apply_crd(self, crd):
+        """
+        Apply the CRD
+
+        Args:
+            crd (str): Name of file to apply
+        """
+        self.crd = crd
+        self.dir += '/ripsaw'
+        run(f'oc apply -f deploy', shell=True, check=True, cwd=self.dir)
+        run(f'oc apply -f {crd}', shell=True, check=True, cwd=self.dir)
+        run(f'oc apply -f {self.operator}', shell=True, check=True, cwd=self.dir)
+
+    def setup_postgresql(self):
+        """
+        Deploy postgres sql server
+        """
+        try:
+            pgsql_service = templating.load_yaml_to_dict(
+                constants.PGSQL_SERVICE_YAML
+            )
+            pgsql_cmap = templating.load_yaml_to_dict(
+                constants.PGSQL_CONFIGMAP_YAML
+            )
+            pgsql_sset = templating.load_yaml_to_dict(
+                constants.PGSQL_STATEFULSET_YAML
+            )
+            self.pgsql_service = OCS(**pgsql_service)
+            self.pgsql_service.create()
+            self.pgsql_cmap = OCS(**pgsql_cmap)
```

Comment:
don't you need to set the kind as weel?

Reply:
the kind exists - its coming from yaml (load_to_yaml)